### PR TITLE
New permissions managing project UI

### PR DESCRIPTION
**Pull request by ZmagoD**
Jira ticket: [SCI-5645](https://biosistemika.atlassian.net/browse/SCI-5645)

### What was done

- added the logic to assign user as an owner when the project is created
- add the CTA for manage permission modal
- added logic/ui to manage users on project
- added logic/ui to add new users + user_assignments to project
- added new API endpoints for task and experiment user_assignments

### CODE REVIEW

**Comment by ZmagoD** [comment]
@okriuchykhin this is ready for code review 👀

File: app/assets/javascripts/shared/autosave_form.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(function() {
+  'use strict';
+
+  function initAutosaveListeners() {
+    $(document).on('change', 'form[data-action*="autosave-form"]', function({ currentTarget }) {
+      $.rails.fire($(currentTarget), 'submit');
+    });
+  }
+
+  $(document).one('turbolinks:load', initAutosaveListeners);
```
Comment (okriuchykhin) [inline app/assets/javascripts/shared/autosave_form.js:10]: 'one', is it a typo?

File: config/initializers/extends/permission_extends.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ module ProjectPermissions
       CREATE_COMMENTS
       MANAGE_COMMENTS
       MANAGE_TAGS
+      MANAGE_ACCESS
```
Comment (okriuchykhin) [inline config/initializers/extends/permission_extends.rb:15]: Do we actually need it alongside with MANAGE permission?

File: config/initializers/extends.rb [hunk 3]
```diff
@@ -381,7 +381,9 @@ class Extends
     create_project_folder: 159,
     move_project_folder: 160,
     rename_project_folder: 161,
-    delete_project_folder: 162
+    delete_project_folder: 162,
+    change_user_role_on_experiment: 163,
```
Comment (okriuchykhin) [inline config/initializers/extends.rb:385]: 163 and 164 are already taken

File: app/views/projects/index/_project_actions_dropdown.html.erb [hunk 4]
```diff
@@ -48,6 +48,17 @@
           <% end %>
         </li>
       <% end %>
+      <!-- Project members access -->
+      <% if can_read_project?(project) %>
+        <li class="form-dropdown-item">
+          <%= link_to can_manage_project_access?(project) ? edit_access_permissions_project_path(project, format: :json) : access_permissions_project_path(project, format: :json),
```
Comment (okriuchykhin) [inline app/views/projects/index/_project_actions_dropdown.html.erb:54]: doesn't it default to json?

File: config/initializers/extends/permission_extends.rb (hunk 2 above)
Comment (ZmagoD) [reply config/initializers/extends/permission_extends.rb:15]: @okriuchykhin this was decided by @radiokills to make that permissions more granular

File: config/initializers/extends.rb (hunk 3 above)
Comment (ZmagoD) [reply config/initializers/extends.rb:385]: @okriuchykhin where are they used I can't find them. Is OK if I use `165` and `166`

File: app/views/projects/index/_project_actions_dropdown.html.erb (hunk 4 above)
Comment (ZmagoD) [reply app/views/projects/index/_project_actions_dropdown.html.erb]: @okriuchykhin can you merge the master in you `biosistemika:features/new_permissions` branch? I'm not able to push directly

**Pull request merged**